pkg/netutil/netutil.go

```diff
@@ -211,3 +214,88 @@ func NerdctlLabels(b []byte) *map[string]string {
 	}
 	return ncl.NerdctlLabels
 }
+
+func GetBridgeName(id int) string {
+	return fmt.Sprintf("nerdctl%d", id)
+}
+
+func GenerateCNIPlugins(driver string, id int, ipam map[string]interface{}) ([]CNIPlugin, error) {
+	if driver == "" {
+		driver = DefaultCNIPlugin
+	}
+	var plugins []CNIPlugin
+	switch driver {
+	case "bridge":
+		bridge := newBridgePlugin(GetBridgeName(id))
+		bridge.IPAM = ipam
+		bridge.IsGW = true
+		bridge.IPMasq = true
+		bridge.HairpinMode = true
+		plugins = []CNIPlugin{bridge, newPortMapPlugin(), newFirewallPlugin(), newTuningPlugin()}
+	case "nat":
+		nat := newNatPlugin("Ethernet")
+		nat.IPAM = ipam
+		plugins = []CNIPlugin{nat}
+	default:
+		return nil, fmt.Errorf("unsupported cni driver %q", driver)
+	}
+	return plugins, nil
+}
+
+func GenerateIPAM(driver string, subnetStr string) (map[string]interface{}, error) {
+	if driver == "" {
+		driver = DefaultIPAMDriver
+	}
+	subnetIP, subnet, err := net.ParseCIDR(subnetStr)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse subnet %q", subnet)
+	}
+	if !subnet.IP.Equal(subnetIP) {
+		return nil, fmt.Errorf("unexpected subnet %q, maybe you meant %q?", subnetStr, subnet.String())
+	}
+	gateway := make(net.IP, len(subnet.IP))
+	copy(gateway, subnet.IP)
+	gateway[3] += 1
+
+	var ipamConfig interface{}
+	switch driver {
+	case "host-local":
+		ipamConf := newHostLocalIPAMConfig()
+		ipamConf.Routes = []IPAMRoute{
+			{Dst: "0.0.0.0/0"},
+		}
+		ipamConf.Ranges = append(ipamConf.Ranges, []IPRange{
+			{
+				Subnet:  subnet.String(),
+				Gateway: gateway.String(),
+			},
+		})
+		ipamConfig = ipamConf
+	case "windows":
+		ipamConf := newWindowsIPAMConfig()
+		ipamConf.Subnet = subnet.String()
+		ipamConf.Routes = append(ipamConf.Routes, IPAMRoute{Gateway: gateway.String()})
+		ipamConfig = ipamConf
+	default:
+		return nil, fmt.Errorf("unsupported ipam driver %q", driver)
+	}
+
```

Comment:
I think that cases are not balanced. `Windows` is an os and not a driver.

you can make `netutil_windows.go` and `netutil_linux.go`